index.js

```diff
@@ -0,0 +1,66 @@
+
+function checkSalaryTotal(lineup) {
+
+  return (45000 > lineup.map((lineup) => lineup.salary).reduce((total, salary) => total + salary, 0))
+}
+
+function checkTeams(lineup) {
+  const teams = lineup.map((lineup) => lineup.teamId)
+  const sortedTeams = teams.sort((a, b) => a - b)
+  let currentCount = 1
+  let highestSteak = 1
+
+  for (let i = 0; i < sortedTeams.length; i++) {
+    if (sortedTeams[i] === sortedTeams[i + 1]) {
+      currentCount++
+    } else {
+      currentCount = 1
+    }
+    if (currentCount > highestSteak) {
+      highestSteak = currentCount
+    }
+  }
+
+  return highestSteak <= 2
+}
+
+function checkGames(lineup) {
+  const games = lineup.map((lineup) => lineup.gameId)
+  const sortedGames = games.sort((a, b) => a - b)
+  let currentCount = 1
+  let highestSteak = 1
+
+  for (let i = 0; i < sortedGames.length; i++) {
+    if (sortedGames[i] === sortedGames[i + 1]) {
+      currentCount++
+    } else {
+      currentCount = 1
+    }
+    if (currentCount > highestSteak) {
+      highestSteak = currentCount
+    }
+  }
+
+  return highestSteak <= 3
+}
```

Comment:
This could be done with a reduce to reformat and then a check which would lessen the number of loops you've got here. You're looping over the lineup 3 times with the map, sort, and for.